pkg/controllers/federatedcluster/clusterstatus.go

```diff
@@ -91,7 +256,11 @@ func (c *FederatedClusterController) collectIndividualClusterStatus(
 	cluster = cluster.DeepCopy()
 	conditionTime := metav1.Now()
 
-	offlineStatus, readyStatus := checkReadyByHealthz(ctx, discoveryClient)
+	timeout := c.clusterHealthCheckConfig.Period / 2
```

Comment:
Origin timeout is 30s. Now it seems the default value is 15s. Is this a `Best practices` that we use half of clusterHealthCheckConfig.Period?

Reply:
30 seconds is actually the experience value I set before. I checked some recent statistics: normally (more than 95%) it takes no more than 10 seconds to pass the health check, and more than 10 seconds usually means that the cluster is disconnected. So I shortened it. 

On the other hand, `clusterHealthCheckConfig` is now configurable, and the minimum is currently 15 seconds (according to feedback, some people want to shorten this interval when katalyst is enabled). It feels inappropriate if the timeout is still fixed at 30s.